ios/AdyenPayment.swift

```diff
@@ -52,8 +52,19 @@ class AdyenPayment: RCTEventEmitter {
         PaymentsData.countryCode = paymentDetails["countryCode"] as! String
         PaymentsData.returnUrl = paymentDetails["returnUrl"] as! String
         PaymentsData.regionId = paymentDetails["regionId"] as! Int
-        PaymentsData.baseRate = paymentDetails["baseRate"] as? Int
-        PaymentsData.effectiveRate = paymentDetails["effectiveRate"] as? Int
+        
+        /// A conversion like this `PaymentsData.baseRate = paymentDetails["baseRate"] as? Int` results in `PaymentsData.baseRate` being set to `0` if the value is `nil`. We want it to be nil if it's not in the dictionary.
+        if let baseRateObject = paymentDetails["baseRate"], let baseRate = baseRateObject as? Int {
+            PaymentsData.baseRate = baseRate
+        } else {
+            PaymentsData.baseRate = nil
```

Comment:
maybe `PaymentsData.baseRate = (paymentDetails["baseRate"] as? Int)?.flatMap(Int.init)` ?

Agree its a bit terse, but if you have to repeat this pattern it becomes a bit more readable IMO.

Reply:
That actually doesn't compile. `Int` doesn't have `flatMap`. It's a moot point now anyway, I changed it back. I thought the conversion was wrong but lldb was giving me bad info.

Reply:
`Int?` does though, so I think the `?` before flatMap didn't need to be there.

Reply:
> `Int?` does though, so I think the `?` before flatMap didn't need to be there.

Correct, because `Optional` has `flatMap`